test/abstract_syntax_tree.coffee

```diff
@@ -1775,22 +1775,109 @@ test "AST as expected for Elision node", ->
       type: 'StringLiteral'
       value: 'asdfqwer'
 
-# test "AST as expected for While node", ->
-#   testExpression 'loop 1',
-#     type: 'While'
-#     condition:
-#       type: 'BooleanLiteral'
-#       value: 'true'
-#       originalValue: 'true'   # TODO: This should probably be changed for Prettier.
-#     body:
-#       type: 'Value'
+test "AST as expected for While node", ->
+  testStatement 'loop 1',
+    type: 'WhileStatement'
+    test:
+      type: 'BooleanLiteral'
+      value: true
+    body:
+      type: 'BlockStatement'
+      body: [
+        type: 'ExpressionStatement'
+        expression: NUMBER 1
+      ]
+    guard: null
+    inverted: no
+    postfix: no
+    loop: yes
 
-#   testExpression 'while 1 < 2 then',
-#     type: 'While'
-#     condition:
-#       type: 'Op'
-#     body:
-#       type: 'Block'
+  testStatement 'while 1 < 2 then',
```

Comment:
Should we add another test where there’s some logic after the `then`? Like `while 1 < 2 then fn()`?

Reply:
Sure, added to AST and location data tests